test/index.jl

```diff
@@ -178,4 +178,53 @@ end
     @test isempty(fuzzymatch(i.lookup, :abcd))
 end
 
+@testset "Regex indexing" begin
+    i = Index()
+    push!(i, :x1)
+    push!(i, :x12)
+    push!(i, :x131)
+    push!(i, :y13)
+    push!(i, :yy13)
+    @test i[r"x1."] == [2, 3]
+    @test i[r"xx"] == Int[]
```

Comment:
```suggestion
    @test i[r"xx"] == []
```

It's a bit misleading since the type isn't checked. Same below.